contrib/rpm/polybar.spec

```diff
@@ -1,70 +1,96 @@
-# optional dependency
-BuildRequires:  pkgconfig(alsa)
-# main dependency
-BuildRequires:  pkgconfig(cairo)
-BuildRequires:  pkgconfig(jsoncpp)
-BuildRequires:  pkgconfig(libcurl)
-BuildRequires:  pkgconfig(libpulse)
-BuildRequires:  pkgconfig(python)
-BuildRequires:  pkgconfig(xcb)
-BuildRequires:  pkgconfig(xcb-proto)
-BuildRequires:  pkgconfig(xcb-util)
+Group:         System/GUI/Other
+License:       MIT
+URL:           https://github.com/jaagr/%{name}	
+Source0:       %{url}/releases/download/%{version}/%{name}-%{version}.tar
+
+BuildRequires: clang >= 3.4
+BuildRequires: cmake >= 3.1
+BuildRequires: git
+BuildRequires: pkgconfig(cairo)
+BuildRequires: pkgconfig(xcb)
+BuildRequires: pkgconfig(xcb-util)
+BuildRequires: pkgconfig(xcb-proto)
+BuildRequires: pkgconfig(xcb-image)
+BuildRequires: pkgconfig(xcb-ewmh)
+BuildRequires: pkgconfig(xcb-xrm)
+BuildRequires: pkgconfig(xcb-cursor)
+BuildRequires: pkgconfig(alsa)
+BuildRequires: pkgconfig(libpulse)
+BuildRequires: pkgconfig(jsoncpp)
+BuildRequires: pkgconfig(libcurl)
+BuildRequires: pkgconfig(libmpdclient)
+BuildRequires: pkgconfig(libnl-genl-3.0)
+
+%if 0%{?fedora}
+BuildRequires: ninja-build
+BuildRequires: python2
+BuildRequires: wireless-tools-devel
+BuildRequires: i3-ipc
+%else
+BuildRequires: ninja
+BuildRequires: pkgconfig
+BuildRequires: python-xml
+BuildRequires: pkgconfig(python)
 %if 0%{?suse_version} <= 1315
-BuildRequires:  i3-devel
+BuildRequires: i3-devel
 %else
-BuildRequires:  i3-gaps-devel
+BuildRequires: i3-gaps-devel
 %endif
 %if 0%{?suse_version}
-BuildRequires:  libiw-devel
+BuildRequires: libiw-devel
 %else
-BuildRequires:  wireless-tools-devel
+BuildRequires: wireless-tools-devel
+%endif
 %endif
 
 %description
-A fast and easy-to-use status bar for tilling WM
+Polybar aims to help users build beautiful and highly customizable status bars for their desktop environment, without the need of having a black belt in shell scripting.
+
+The main purpose of Polybar is to help users create awesome status bars. It has built-in functionality to display information about the most commonly used services.
 
 %prep
-%setup -q
+%setup -q -n %{name}
+git config --global http.sslVerify false
+mkdir -p %{_target_platform}
 
 %build
-%cmake
-make
+pushd %{_target_platform}
+    cmake -G Ninja \
+    -DCMAKE_CXX_COMPILER=clang++ \
+    -DCMAKE_INSTALL_PREFIX:PATH=%{_prefix} \
+    -DBUILD_TESTS:BOOL=ON \
+    -DCMAKE_BUILD_TYPE=Debug \
+    ..
+popd
+%ninja_build -C %{_target_platform}
 
 %install
-%cmake_install
+%ninja_install -C %{_target_platform}
+
+%check
+pushd %{_target_platform}
+    ctest --output-on-failure
+popd
 
 %files
-%dir %{_datadir}/bash-completion/
-%dir %{_datadir}/bash-completion/completions
-%dir %{_datadir}/doc/%{name}
-%dir %{_datadir}/zsh/
-%dir %{_datadir}/zsh/site-functions
+%doc README.md
+%license LICENSE
+
 %{_bindir}/%{name}
 %{_bindir}/%{name}-msg
-%{_datadir}/doc/%{name}/config
-%{_mandir}/man1/%{name}.1%{?ext_man}
+%{_mandir}/man1/%{name}.1.gz
```

Comment:
no man macro on fedora?

Reply:
It is an openSuse's macro.

Reply:
If Fedora doesn't have this macro then let's use your way here, since it works for both.